bigquery/cloud-client/src/main/java/com/example/bigquery/QueryParametersSample.java

```diff
@@ -0,0 +1,275 @@
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.example.bigquery;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.QueryRequest;
+import com.google.cloud.bigquery.QueryResponse;
+import com.google.cloud.bigquery.QueryResult;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A sample that demonstrates use of query parameters.
+ */
+public class QueryParametersSample {
+  private static final int ERROR_CODE = 1;
+
+  private static void printUsage() {
+    System.err.println("Usage:");
+    System.err.printf(
+        "mvn exec:java -Dexec.mainClass=%s -Dexec.args=%s\n",
+        QueryParametersSample.class.getCanonicalName(),
+        "sample");
+    System.err.println("sample values: named|array|struct|timestamp");
+    System.err.println("Usage for sample=named:");
+    System.err.println("\tnamed corpus minWordCount");
+    System.err.println("Usage for sample=array:");
+    System.err.println("\tarray gender states...");
+    System.err.println("\tgender=M|F");
+    System.err.println("\tstates=Upper-case 2-letter code for U.S. state, e.g. CA.");
+  }
+
+  /**
+   * Prompts the user for the required parameters to perform a query.
+   */
+  public static void main(final String[] args) throws IOException, InterruptedException {
+    if (args.length < 1) {
+      System.err.println("Expected first argument 'sample'");
+      printUsage();
+      System.exit(ERROR_CODE);
+    }
+    String sample = args[0];
+
+    switch (sample) {
+      case "named":
+        if (args.length != 3) {
+          System.err.println("Unexpected number of arguments for named query sample.");
+          printUsage();
+          System.exit(ERROR_CODE);
+        }
+        runNamed(args[1], Long.parseLong(args[2]));
+        break;
+      case "array":
+        if (args.length < 2) {
+          System.err.println("Unexpected number of arguments for array query sample.");
+          printUsage();
+          System.exit(ERROR_CODE);
+        }
+        String gender = args[1];
+        ArrayList<String> states = new ArrayList<>();
+        for (int i = 2; i < args.length; i++) {
+          states.add(args[i]);
+        }
+        runArray(gender, states);
```

Comment:
I would just to `Arrays.copyOfRange(args, 2, args.length)` and pass around a `String[]` rather than converting to and from lists. This has the added benefit of the `runArray` method taking an array as an argument instead of a list :-)